Assets/_BForBoss/_Entities/Scripts/LifeCycleManager.cs

```diff
@@ -1,19 +1,23 @@
 using UnityEngine;
+using System;
 using System.Linq;
 
 namespace Perigon.Entities
 {
     public class LifeCycleManager : MonoBehaviour
     {
+        public Action<int> OnLivingEntityEliminated;
         private LifeCycleBehaviour[] _lifeCycleBehaviours = null;
-
+        private int _totalEnemiesEliminated = 0;
+        
         public int LivingEntities => _lifeCycleBehaviours.Count(life => life.IsAlive);
 
-        private void Awake()
+        public void EliminateLivingEntity()
```

Comment:
hmm so this is `public` because each entity needs to be able to call this right? 
So from an outside perspective, `EleminateLivingEntity()` might give off the meaning of.. Okay this manager is going to kill off a random entity? 

But if you call this, nothing really happens. So it can be misleading.

Reply:
So what you can do instead. 

Is to pass a lambda function into the entity instead. That way, they don't need to know about the manager, and they just need to run whatever lambda is passed through when they die